### DISCUSSION TITLE
Appropriate polyfills for IE11

### DISCUSSION THREAD

**Discussion by kwht-k** [react-hook-form/react-hook-form#2468] [Q&A] [answered]
I'm trying to use React Hook Form on IE11.
I referred to [FAQs#BrowserSupport](https://react-hook-form.com/faqs#BrowserSupport), but I got some errors shown below instead of the error "Object doesn't support property or method find".

![img1_missing_value](https://user-images.githubusercontent.com/54492352/89252063-a5c4f700-d653-11ea-9243-acd4703265b4.png)
![img2_missing_entries](https://user-images.githubusercontent.com/54492352/89252072-abbad800-d653-11ea-9a1d-2b2fbcc0ceaa.png)

Therefore, I added the import statements shown below, and this code worked fine.

#### App.js
```javascript
import 'react-app-polyfill/ie11'; // addition
import 'core-js/stable/object/values'; // addition
import 'core-js/stable/object/entries'; // addition
import React from 'react';
import ReactDOM from 'react-dom';
import { useForm } from 'react-hook-form/dist/index.ie11';

function Form() {
  const { register, formState, handleSubmit } = useForm({
    mode: 'onBlur',
  });

  const onSubmit = (data) => {
    alert(JSON.stringify(data));
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>Name</label>
      <input type="text" name="name" ref={register({ required: true })} />
      <input type="submit" disabled={!formState.isValid} value="Submit" />
    </form>
  );
}

function App() {
  const mount = () => {
    ReactDOM.render(<Form />, document.getElementById('form-area'));
  };

  const unmount = () => {
    ReactDOM.unmountComponentAtNode(document.getElementById('form-area'));
  };

  return (
    <div>
      <button type="button" onClick={mount}>
        Mount
      </button>
      <button type="button" onClick={unmount}>
        UnMount
      </button>
      <div id="form-area"></div>
    </div>
  );
}

export default App;
```

Could you please tell me this is correct usage or not?

In addition,I tried `react-app-polyfill/stable`(cf. [Polyfilling other language features](https://github.com/facebook/create-react-app/tree/master/packages/react-app-polyfill#polyfilling-other-language-features). It worked but caused a memory leak when the components were unmounted(Please refer to the screenshots below).
So I want to know which polyfills React Hook Form needs precisely.

#### Detected Memory leak
These screenshots were taken using the following steps.

1. Start the "profilling session" in IE11
2. Push the "Mount" button to mount the components.
3. Push the "Unmount" button to unmount the components.
4. Execute "Take heap snapshot".
5. Repeat `step2` to `step 4`.

![img3_leak](https://user-images.githubusercontent.com/54492352/89252109-c8571000-d653-11ea-89af-73cb90b6a8da.png)
![img4_no-leak](https://user-images.githubusercontent.com/54492352/89252114-cdb45a80-d653-11ea-9646-27131664acb8.png)

#### Environment
- OS: Windows10 64bit
- Browser: Internet Explorer 11 (11.959.18362.0)
- Version of React Hook Form: 6.2.0
- Using Create React App v3.4.1

**Comment by bluebill1049** [comment]
hmmm strange, ie11 build should work without other polyfills. 

https://github.com/react-hook-form/react-hook-form/blob/master/rollup/rollup.ie11.config.js

**Comment by yhor1e** [comment, marked as answer]
@kwht-k 

I created #2489. And merged version [v6.3.1](https://github.com/react-hook-form/react-hook-form/releases/tag/v6.3.1) was released.
Maybe you should use v6.3.1.
